### ISSUE TITLE
How to change sys.path?

### ISSUE THREAD

**Issue by ubaldot** [ubaldot/vim-conda-activate#4]
Changing `sys.path` as done before makes omnicomplete (`<c-x><c-o>`) to crash, in-spite everything seemed to be ok. 
Figure out how the internal Vim `sys.path` shall be changed. At the moment, the part of code related to the internal sys.path is commented out.

**Comment by Konfekt** [comment]
Is this plug-in, as its name suggests, more about activating or switching an environment?
Since an environment has to be activated before Vim is entered, I wonder what `:CondaActivate` achieves?
Is it?

> The internal sys.path used by Vim is also set.

This seems from a user's point of view important for omni-completion using the libraries of the current environment.

Could the environment on entering Vim be auto-detected to adapt the (missing) Vim settings?

**Comment by ubaldot** [comment]
>Is this plug-in, as its name suggests, more about activating or switching an environment?
>Since an environment has to be activated before Vim is entered, I wonder what :CondaActivate achieves?
>Is it?

>>The internal sys.path used by Vim is also set.

Yes, exactly. It change the runtime. When you conda activate you don't just change the internal Vim, `sys.path` but you also change the shell `$PATH` and a number of other environment variables.  

>Could the environment on entering Vim be auto-detected to adapt the (missing) Vim settings?

In reality the setting is not missing, it is set to a certain value.
I wanted to auto-detect the environment, but for some reason it does not work. But it is also true that when I wrote this plugin my Vim scripting skills were very lower compared to now.

What I have observed is that some plugins don't seem to like when the runtime is changed dynamically, (see e.g. https://github.com/yegappan/lsp/issues/338). 

Anyways, PR:S are welcome :)

**Comment by Konfekt** [comment]
Thank you very much!

> In reality the setting is not missing, it is set to a certain value.

I hesitated writing `(missing)`, but it was meant that all the shell environment variables, such as `$PATH`, should already have been taken care of by the conda activate command in the shell before starting Vim, which is required to make `:CondaActivate` work; so the question was rather if`sys.path` is the only missing setting that `:CondaActivate` must handle since the shell command cannot?

> I wanted to auto-detect the environment, but for some reason it does not work.

Could you just quickly tell which command you expected to show it, but didn't?

**Comment by ubaldot** [comment]
Btw, by looking at `conda info --json` one can see what is changed when a virtual environment is changed.

**Comment by Konfekt** [comment]
In this [discussion](https://github.com/ubaldot/vim-conda-activate/issues/4) I didn't grasp why the counterpart for conda needed to be inside the environment before. It seems that the (https://github.com/Coacher/vim-virtualenv/) plug-in solved a similar problem without that need. Maybe @Coacher could give a hint if this is really necessary.

**Comment by ubaldot** [comment]
Oh that is interesting! Thanks for pointing it out! 

I think that conda is more complex than virtualenv, and perhaps there are some nuances that I failed to grasp. Consider, for example, that When you install conda a bunch of things are also added to e.g. .bashrc. 

This was one of my first plugins, and as said there is room for refactoring. PRs that include that or any other improvements are welcome! :D
